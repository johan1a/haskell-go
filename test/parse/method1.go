package main
import "fmt";

type circle struct {
    radius int
}

func (c circle) diameter() int {
    return c.radius * 2
}

func main() {
}
