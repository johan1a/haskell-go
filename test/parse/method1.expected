SourceFile (Package "main") [ImportDecl [I1 "fmt"]] [TopLevelDecl1 (TypeDecl [TypeSpec "circle" (TypeLit (StructTypeLit (StructType1 [FieldDecl1 [IdDecl "radius"] (TypeName (TypeNameIdentifier "int"))])))]),TopLevelDecl3 (MethodDecl1 [ParameterDecl2 [IdDecl "c"] (TypeName (TypeNameIdentifier "circle"))] "diameter" (Signature2 [] (Result2 (TypeName (TypeNameIdentifier "int")))) (Block [ReturnStmt (BinExpr (AritmExpr (MulExpr (UnaryExpr (PrimaryExpr (PrimaryExpr1 (Operand2 (OperandName2 (QualifiedIdent "c" "radius")))))) (UnaryExpr (PrimaryExpr (PrimaryExpr1 (Operand1 (BasicLit (IntLit 2)))))))))])),TopLevelDecl2 (FunctionDecl2 "main" (Signature1 []) (Block []))]
